llms/mlx_lm/convert.py

```diff
@@ -183,12 +174,16 @@ def convert(
     upload_repo: str = None,
 ):
     print("[INFO] Loading")
-    weights, config, tokenizer = fetch_from_hub(hf_path)
+    model, config, tokenizer = fetch_from_hub(hf_path)
+
+    weights = dict(tree_flatten(model.parameters()))
     dtype = mx.float16 if quantize else getattr(mx, dtype)
     weights = {k: v.astype(dtype) for k, v in weights.items()}
+
     if quantize:
         print("[INFO] Quantizing")
-        weights, config = quantize_model(weights, config, hf_path, q_group_size, q_bits)
+        model.load_weights(list(weights.items()))
```

Comment:
Reload the weights to ensure that the model's weights are converted to fp16 for better quantization.